snuba/web/split.py

```diff
@@ -128,9 +173,51 @@ def time_split(dataset, request: Request, *args, **kwargs):
 
         return overall_result
 
-    def col_split(
-        dataset, request: Request, column_split_spec: ColumnSplitSpec, *args, **kwargs
-    ):
+
+class ColumnSplitSpec(NamedTuple):
+    """
+    Provides the column names needed to perform column splitting.
+    id_column represent the identity of the row.
+
+    """
+
+    id_column: str
+    project_column: str
+    timestamp_column: str
+
+    def get_min_columns(self) -> Sequence[str]:
+        return [self.id_column, self.project_column, self.timestamp_column]
+
+
+class ColumnSplitQueryStrategy(StorageQuerySplitStrategy):
+    """
+    A strategy that performs column based splitting: if the client requests enough columns,
+    a first query on the minimum set of columns is ran to load as little Clickhouse data
+    as possible. A second query based on the results of the first is then executed to
+    build the full result set.
+    """
+
+    def __init__(self, spec: ColumnSplitSpec) -> None:
+        self.__column_split_spec = spec
+
+    def can_execute(self, request: Request) -> bool:
+        if not _is_query_splittable(request.query):
+            return False
+
+        total_col_count = len(request.query.get_all_referenced_columns())
+        copied_query = copy.deepcopy(request.query)
+        copied_query.set_selected_columns(self.__column_split_spec.get_min_columns())
+        min_col_count = len(copied_query.get_all_referenced_columns())
+
+        selected_cols = request.query.get_selected_columns()
+        return (
+            selected_cols is not None
+            and len(selected_cols) > 0
+            and not request.query.get_aggregations()
```

Comment:
This all seems like it should be checked before the `copy` and early return, foregoing the copy if not required.